drivers/platform/stm32/stm32_gpio_irq.c

```diff
@@ -83,28 +83,50 @@ static bool initialized[STM32_IRQ_CTRL_NB] =  {false};
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 /**
- * @brief EXTI GPIO Interrupt handler callback
+ * @brief Generic Interrupt handler callback
  * @param pin pin number on which the interrupt occurred (GPIO_PIN_pin)
  */
-void HAL_GPIO_EXTI_Callback(uint16_t pin)
+static void stm32_handle_generic_callback(uint16_t pin)
 {
 	int ret;
 	struct irq_action *action;
 	struct irq_action key;
 
 	key.irq_id = no_os_find_first_set_bit(pin);
 
-	if (__HAL_GPIO_EXTI_GET_IT(pin)) {
```

Comment:
is this removed by mistake ?

Reply:
no, it is not removed by mistake. until now HAL_GPIO_EXTI_Callback was called directly from EXTI IRQ handler (due to the rename in stm32.mk script). Now, HAL_GPIO_EXTI_IRQHandler is called instead which is taking care of checking and clearing the interrupt.